tests/initcoder_test.go

```diff
@@ -0,0 +1,157 @@
+	t.Run("create transaction with enough gas for init code", func(t *testing.T) {
+		// we need to provide same gas to all transactions so that we can compare the cost.
+		cost1Word, _ := createAndCheckContractWithCodeLenAndGas(require, net, contract, variant, 30, gasFor3Words)
+		cost2Words, _ := createAndCheckContractWithCodeLenAndGas(require, net, contract, variant, 42, gasFor3Words)
+		cost3Words, _ := createAndCheckContractWithCodeLenAndGas(require, net, contract, variant, 90, gasFor3Words)
+		require.Equal(cost2Words-cost1Word, wordCost,
+			"cost difference between 1 and 2 words should be %v, instead got %d", wordCost, cost2Words-cost1Word)
+		require.Equal(cost3Words-cost1Word, gasCostFor2Words,
+			"cost difference between 1 and 3 words should be %v, instead got %d", gasCostFor2Words, cost3Words-cost1Word)
+	})
+
+	t.Run("create transaction without enough gas for init code", func(t *testing.T) {
+		receipt = createContractWithCodeLenAndGas(require, net, variant, 30, gasForContract-wordCost)
+		require.Equal(types.ReceiptStatusFailed, receipt.Status,
+			"unexpectedly succeeded to create contract without enough gas")
+	})
+
+	t.Run("create transaction with max init code size", func(t *testing.T) {
+		_, _ = createAndCheckContractWithCodeLenAndGas(require, net, contract, variant, MAX_INIT_CODE_SIZE, sufficientGas)
+	})
+
+	t.Run("create transaction with MAX_INITCODE_SIZE+1", func(t *testing.T) {
+		receipt = createContractWithCodeLenAndGas(require, net, variant, MAX_INIT_CODE_SIZE+1, sufficientGas)
+		require.Equal(types.ReceiptStatusFailed, receipt.Status,
+			"unexpectedly succeeded to create contract with init code length greater than MAX_INITCODE_SIZE")
+	})
+}
+
+func createAndCheckContractWithCodeLenAndGas(require *require.Assertions, net *IntegrationTestNet, contract *initcode.Initcode, variant variant, codeLen, gasLimit uint64) (uint64, uint64) {
+	receipt := createContractWithCodeLenAndGas(require, net, variant, codeLen, gasLimit)
+	require.Equal(types.ReceiptStatusSuccessful, receipt.Status, "failed to create contract with code length %d", codeLen)
+	log, err := contract.ParseLogCost(*receipt.Logs[0])
+	require.NoError(err)
+	return log.Cost.Uint64(), receipt.GasUsed
+}
+
+func createContractWithCodeLenAndGas(require *require.Assertions, net *IntegrationTestNet, variant variant, codeLen, gasLimit uint64) *types.Receipt {
+	receipt, err := net.Apply(func(opts *bind.TransactOpts) (*types.Transaction, error) {
+		opts.GasLimit = gasLimit
+		return variant(opts, big.NewInt(int64(codeLen))) //
+	})
+	require.NoError(err)
+	return receipt
+}
+
+type variant func(opts *bind.TransactOpts, codeSize *big.Int) (*types.Transaction, error)
+
+func runTransactionWithCodeSizeAndGas(r *require.Assertions, net *IntegrationTestNet, codeSize, gas uint64) (*types.Receipt, error) {
+	// these values are needed for the transaction but are irrelevant for the test
+	client, err := net.GetClient()
+	r.NoError(err, "failed to connect to the network:")
```

Comment:
The connection to the client should be closed. 
Just like other uses inside the integration net API.

defer client.Close()

Reply:
I closed it manually in line 144, because the call to `net.Run` in line 156 internally makes another call to `GetClient` and this sometimes resulted in a timeout. closing the client before calling run, stopped the timeouts. But I have to remove the commented defer, this was an oversight, sorry.